spark/src/test/scala/geotrellis/spark/summary/polygonal/MaxDoubleSpec.scala

```diff
@@ -16,18 +16,32 @@
 
 package geotrellis.spark.summary.polygonal
 
+import cats.Monoid
 import geotrellis.spark._
-import geotrellis.spark.io.hadoop._
-import geotrellis.spark.testkit.testfiles._
+import geotrellis.spark.testkit.testfiles.TestFiles
 import geotrellis.raster.summary.polygonal._
 import geotrellis.spark.testkit._
-
 import geotrellis.raster._
+import geotrellis.raster.summary.polygonal.visitors.MaxVisitor
 import geotrellis.vector._
-
 import org.scalatest.FunSpec
 
-class MaxDoubleSpec extends FunSpec with TestEnvironment with TestFiles {
+class MaxDoubleSpec extends FunSpec with TestEnvironment with TestFiles with PolygonalUtils {
+
+  implicit def maxMonoid: Monoid[Option[Double]] = new Monoid[Option[Double]] {
```

Comment:
Each other test type is going to need its own monoid implementation. Should we bundle these somewhere under `geotrellis.spark.summary.polygonal` so that users can take advantage of a few default implementations for max, min, sum and mean? I'd make the argument that we should, in which case I guess they'd live under `geotrellis.spark.summary.polygonal.Implicits`? That's going to be weird though because well have multiple implicits with the same type signature, i.e. monoids for `Option[Double]`

Reply:
You get all of the standard ones from cats, but the weird part is that the meaning of mooned is more specific to type of summary here. `combine` is going to be different for `MaxVisitor` vs `MinVisitor`

```scala
scala> import cats.implicits._
import cats.implicits._

scala> import cats._
import cats._

scala> Monoid[Option[Double]]
res1: cats.kernel.Monoid[Option[Double]] = cats.kernel.instances.OptionMonoid@702e6ef5
```

Reply:
Yeah. I don't really like this at all. If we're going to talk about a type class for `RDD[(K, V)]` and `Seq[(K, V)]` we also should talk about this problem and how it intersects with [your comment](https://github.com/CloudNiner/geotrellis/pull/2#discussion_r294888924) about how we're returning `R` instead of `PolygonalSummaryResult[R]`. It's likely that we could fix both of these problems with a few steps:

- swap to returning `RDD[Feature[Geometry, PolygonalSummaryResult[R]]]` which removes the need for monoid.empty
- require that the user provide a `combine(a: R, b: R): R` method to `RDDPolygonalSummary.apply`, similar to `RDD.reduce` or `RDD.aggregate`
- remove the monoid, since it really isn't one in that it doesn't apply generically for all `T`
- provide a few combine methods in a helper object for some common operations

The only oddity is that combine might have to deal with instances of `PolygonalSummaryResult[R]` instead of `R` which could be awkward, or we could just flatten any `NoIntersection` results we encounter within our implementation.

Reply:
@echeipesh and I discussed this offline. The result was to develop wrapper types for the summary operations that can have monoids and to return `PolygonalSummaryResult[R]` for consistency with the raster version of this API. Moving the monoids to wrapper types prevents having a number of `Monoid[Option[Double]]` in scope that have to be explicitly provided or handled with scoped imports. Ideally in the future we can move these types to [opaque types](https://docs.scala-lang.org/sips/opaque-types.html). Giving a user provided combine function potentially duplicates implementation and causes the user to have to think about the combine operation in two locations -- in the visitor and at the call site of RDDPolygonalSummary. This way, we encapsulate thinking about the combine operation in one location...on the type itself.